src/System.Runtime.Serialization.Xml/tests/CanonicalizationTestHelper.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            return CreateXmlReader(rwType, stream, encoding, quotas, dictionary, null);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, byte[] buffer, Encoding encoding, XmlDictionaryReaderQuotas quotas)
+        {
+            return CreateXmlReader(rwType, buffer, encoding, quotas, null);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, Stream stream, Encoding encoding, XmlDictionaryReaderQuotas quotas)
+        {
+            return CreateXmlReader(rwType, stream, encoding, quotas, null);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, byte[] buffer, Encoding encoding)
+        {
+            return CreateXmlReader(rwType, buffer, encoding, XmlDictionaryReaderQuotas.Max);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, Stream stream, Encoding encoding)
+        {
+            return CreateXmlReader(rwType, stream, encoding, XmlDictionaryReaderQuotas.Max);
+        }
+
+        public static XmlWriter CreateXmlWriter(ReaderWriterType rwType, Stream stream, Encoding encoding)
+        {
+            return CreateXmlWriter(rwType, stream, encoding, null);
+        }
+
+        public static XmlWriter CreateXmlWriter(ReaderWriterType rwType, Stream stream, Encoding encoding, IXmlDictionary dictionary)
+        {
+            XmlWriter result = null;
+            switch (rwType)
+            {
+                case ReaderWriterType.Binary:
+                    result = XmlDictionaryWriter.CreateBinaryWriter(stream, dictionary);
+                    break;
+                case ReaderWriterType.Text:
+                    result = XmlDictionaryWriter.CreateTextWriter(stream, encoding);
+                    break;
+                case ReaderWriterType.MTOM:
+                    result = XmlDictionaryWriter.CreateMtomWriter(stream, encoding, int.MaxValue, "myStartInfo", null, null, true, false);
+                    break;
+                case ReaderWriterType.WebData:
+                    XmlWriterSettings settings = new XmlWriterSettings();
+                    settings.Encoding = encoding;
+                    result = XmlWriter.Create(stream, settings);
+                    break;
+                case ReaderWriterType.WrappedWebData:
+                    XmlWriterSettings settings2 = new XmlWriterSettings();
+                    settings2.Encoding = encoding;
+                    result = XmlWriter.Create(stream, settings2);
+                    result = XmlDictionaryWriter.CreateDictionaryWriter(result);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException("rwType");
+            }
+            return result;
+        }
+    }
+
+    public class Helper
+    {
+        public static void CompareArrays(byte[] array1, int offset1, byte[] array2, int offset2, int count)
```

Comment:
I think we can use `Assert.True(Enumerable.SequenceEqual(array1, array2)` instead of having this method.